src/NovelAudioService.cpp

```diff
@@ -0,0 +1,173 @@
+  if (Mix_AllocateChannels(2) < 0)
+  {
+    std::cout << "ERROR: Failed to allocate channels." << std::endl;
+    isInitialized = false;
+    return -1;
+  }
+  std::cout << "INFO: Mixer engaged!" << std::endl;
+
+  return 1;
+}
+
+void NovelAudioService::loadSound(std::string input) {
+  auto exists = _sounds.find(input);
+  if (exists != _sounds.end()) return;
+
+  Mix_Chunk* newSound = Mix_LoadWAV(input.c_str());
+  if (newSound != nullptr)
+  {
+    _sounds[input] = newSound;
+  }
+
+}
+
+void NovelAudioService::unloadSound(std::string input) {
+  auto existingSound = _sounds.find(input);
+  if (existingSound == _sounds.end()) return;
+
+  Mix_FreeChunk(_sounds[input]);
+  _sounds.erase(existingSound);
+}
+
+void NovelAudioService::loadMusic(std::string input) {
+  auto exists = _music.find(input);
+  if (exists != _music.end()) return;
+
+  Mix_Music* newMusic = Mix_LoadMUS(input.c_str());
+  if (newMusic != nullptr)
+  {
+    _music[input] = newMusic;
+  }
+}
+
+void NovelAudioService::unloadMusic(std::string input) {
+  auto existingMusic = _music.find(input);
+  if (existingMusic == _music.end()) return;
+
+  Mix_FreeMusic(_music[input]);
+  _music.erase(existingMusic);
+}
+
+//Specifying the filename for playMusic() infers that it has not been played
+//or that the music should be switched.
+void NovelAudioService::playMusic(std::string musicName) {
+  auto existingMusic = _music.find(musicName);
+  if (existingMusic == _music.end())
+  {
+    loadMusic(musicName);
+    existingMusic = _music.find(musicName);
+    if (existingMusic == _music.end())
+    {
+      return;
+    }
+  }
+
+  Mix_PlayMusic(_music[musicName],-1);
+}
+
+//Not specifying a filename assumes that the music channel has been paused.
+//We doublecheck to make sure that is the case
+void NovelAudioService::playMusic() {
+  if (Mix_PausedMusic() == 1)
+  {
+    Mix_ResumeMusic();
+  }
+}
+
+void NovelAudioService::playSound(std::string soundName){
+  auto existingSound = _sounds.find(soundName);
+  if (existingSound == _sounds.end())
+  {
+    loadSound(soundName);
+    existingSound = _sounds.find(soundName);
+    if (existingSound == _sounds.end())
+    {
+      return;
+    }
+  }
+
+  auto onChannel = _channelMap.find(soundName);
+  if (onChannel == _channelMap.end())
+  {
+    _channelMap[soundName] = _nextChannel;
+    Mix_PlayChannel(_nextChannel, _sounds[soundName], 0);
+    std::cout << "DEBUG: Channel " << _nextChannel << "loaded." << std::endl;
+    int nextChannelTest = _nextChannel + 1;
+    _nextChannel = (nextChannelTest >= 2 || nextChannelTest < 0) ? -1 : nextChannelTest;
+  }
+  else
+  {
+    if (Mix_Playing(_channelMap[soundName]) == 1)
+    {
+      std::cout << "Already playing on channel " << _channelMap[soundName] << std::endl;
+      return;
+    }
+    Mix_PlayChannel(_channelMap[soundName], _sounds[soundName], 0);
+  }
+
+}
+
+void NovelAudioService::pauseMusic() {
+  if(Mix_PausedMusic() == 0)
+  {
```

Comment:
bracing braeks convention. Also maybe consider inverting or one-lining this?

Reply:
I'll take a look at this and see about one lining it. Should be possible.